### PR TITLE
[enhancement] Added Firebreather trainer class

### PR DESCRIPTION
**Pull request by Alekemon**
2nd out of the 4 trainer class pull requests. This trainer class uses Pokémon who have Smog in their learnset

### CODE REVIEW

**Comment by bennybroseph** [comment]
Can we get a screenshot or video showing this was tested?

**Comment by Alekemon** [comment]
> Can we get a screenshot or video showing this was tested?

image

**Comment by CodeTappert** [comment]
German: "Feuerspucker"

**Comment by bennybroseph** [comment]
Does this have dialogue?

**Comment by Alekemon** [comment]
> Does this have dialogue?

I forgot to re-add dialogue. I'll mark this as a draft and add it tomorrow

**Comment by Alekemon** [comment]
Ok I re-added dialogue here, I'll do the same with the other PRs

File: src/locales/de/dialogue.ts [hunk 1]
```diff
@@ -367,6 +367,18 @@ export const PGMdialogue: DialogueTranslationEntries = {
...
   },
+  "firebreather": {
+    "encounter": {
+      1: "My flames shall devour you!",
+      2: "I'm seriously upset by your cold confusion!",
+      3: "I'm going to put on my fire-breathing shows!"
+    },
+    "victory": {
+      1: "If we only had it warmer.",
+      2: "I flamed out.",
```
Comment (bennybroseph) [inline src/locales/de/dialogue.ts:378]: I couldn't find a source for this:

> If we only had it warmer.
> I flamed out.

Let's use this instead:

> I burned down to ashes...
> Yow! That's hot!

**Review by bennybroseph — CHANGES_REQUESTED** [review]

**Comment by Alekemon** [comment]
Ok, I finished adding the requested dialogue lines. Thanks!

**Review by bennybroseph — APPROVED** [review]

File: src/data/trainer-config.ts [hunk 2]
```diff
@@ -989,6 +989,8 @@ export const trainerConfigs: TrainerConfigs = {
   [TrainerType.DEPOT_AGENT]: new TrainerConfig(++t).setMoneyMultiplier(1.45).setEncounterBgm(TrainerType.CLERK),
   [TrainerType.DOCTOR]: new TrainerConfig(++t).setHasGenders("Nurse", "lass").setHasDouble("Medical Team").setMoneyMultiplier(3).setEncounterBgm(TrainerType.CLERK)
     .setSpeciesFilter(s => !!s.getLevelMoves().find(plm => plm[1] === Moves.HEAL_PULSE)),
+  [TrainerType.FIREBREATHER]: new TrainerConfig(++t).setMoneyMultiplier(1.4).setEncounterBgm(TrainerType.ROUGHNECK)
+    .setSpeciesFilter(s => !!s.getLevelMoves().find(plm => plm[1] === Moves.SMOG)),
```
Comment (bennybroseph) [inline src/data/trainer-config.ts:993]: would like them to also pick some fire pokemon, not just smog pokemon

**Review by bennybroseph — CHANGES_REQUESTED** [review]

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**